File: frontend/src/types/procedure.ts
export interface ProcedureCategory {
  id: string;
  name_ko: string;
  name_en: string | null;
  name_ja: string | null;
  name_zh: string | null;
  slug: string;
  parent_id: string | null;
  sort_order: number;
}

export interface ProcedureCategoryTree extends ProcedureCategory {
  children: ProcedureCategoryTree[];
}

export interface Procedure {
  id: string;
  category_id: string | null;
  name_ko: string;
  name_en: string | null;
  name_ja: string | null;
  name_zh: string | null;
  name_vi: string | null;
  slug: string;
  description_ko: string | null;
  description_en: string | null;
  effects_ko: string | null;
  duration_minutes: number | null;
  effect_duration: string | null;
  downtime_days: number | null;
  min_interval_days: number | null;
  pain_level: number | null;
  anesthesia_options: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string | null;
}

export interface ClinicProcedure {
  id: string;
  clinic_id: string;
  procedure_id: string;
  custom_name_ko: string | null;
  custom_description_ko: string | null;
  custom_effects_ko: string | null;
  custom_duration_minutes: number | null;
  custom_downtime_days: number | null;
  custom_precautions_before: string | null;
  custom_precautions_after: string | null;
  is_active: boolean;
  sales_performance_score: number | null;
  created_at: string;
  updated_at: string | null;
  // Joined
  procedure: Procedure | null;
}

export interface ProcedurePricing {
  id: string;
  clinic_procedure_id: string;
  clinic_id: string;
  regular_price: number;
  event_price: number | null;
  discount_rate: number | null;
  event_start_date: string | null;
  event_end_date: string | null;
  is_package: boolean;
  package_details: Record<string, unknown> | null;
  prices_by_currency: Record<string, number>;
  discount_warning: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string | null;
}

File: backend/app/schemas/procedure.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class ProcedureCreate(BaseModel):
    category_id: uuid.UUID | None = None
    name_ko: str = Field(..., min_length=1, max_length=200)
    name_en: str | None = None
    name_ja: str | None = None
    name_zh: str | None = None
    name_vi: str | None = None
    slug: str = Field(..., min_length=1, max_length=200, pattern=r"^[a-z0-9\-]+$")
    description_ko: str | None = None
    description_en: str | None = None
    effects_ko: str | None = None
    duration_minutes: int | None = None
    effect_duration: str | None = None
    downtime_days: int | None = None
    min_interval_days: int | None = None
    common_side_effects: str | None = None
    rare_side_effects: str | None = None
    dangerous_side_effects: str | None = None
    precautions_before: str | None = None
    precautions_during: str | None = None
    precautions_after: str | None = None
    pain_level: int | None = Field(None, ge=1, le=10)
    pain_type: str | None = None
    anesthesia_options: str | None = None
    anesthesia_details: dict | None = None
    contraindications: dict | None = None


class ProcedureUpdate(BaseModel):
    category_id: uuid.UUID | None = None
    name_ko: str | None = Field(None, min_length=1, max_length=200)
    name_en: str | None = None
    name_ja: str | None = None
    name_zh: str | None = None
    name_vi: str | None = None
    description_ko: str | None = None
    description_en: str | None = None
    effects_ko: str | None = None
    duration_minutes: int | None = None
    effect_duration: str | None = None
    downtime_days: int | None = None
    min_interval_days: int | None = None
    common_side_effects: str | None = None
    rare_side_effects: str | None = None
    dangerous_side_effects: str | None = None
    precautions_before: str | None = None
    precautions_during: str | None = None
    precautions_after: str | None = None
    pain_level: int | None = Field(None, ge=1, le=10)
    pain_type: str | None = None
    anesthesia_options: str | None = None
    anesthesia_details: dict | None = None
    contraindications: dict | None = None
    is_active: bool | None = None


class ProcedureResponse(BaseModel):
    id: uuid.UUID
    category_id: uuid.UUID | None
    name_ko: str
    name_en: str | None
    name_ja: str | None
    name_zh: str | None
    name_vi: str | None
    slug: str
    description_ko: str | None
    description_en: str | None
    effects_ko: str | None
    duration_minutes: int | None
    effect_duration: str | None
    downtime_days: int | None
    min_interval_days: int | None
    common_side_effects: str | None
    rare_side_effects: str | None
    dangerous_side_effects: str | None
    precautions_before: str | None
    precautions_during: str | None
    precautions_after: str | None
    pain_level: int | None
    pain_type: str | None
    anesthesia_options: str | None
    anesthesia_details: dict | None
    contraindications: dict | None = None
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
